File: app/src/main/java/com/example/asiantech/playmusic/adapter/PagerAdapter.java
package com.example.asiantech.playmusic.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.example.asiantech.playmusic.fragment.AlbumsFragment_;
import com.example.asiantech.playmusic.fragment.ArtistFragment_;
import com.example.asiantech.playmusic.fragment.SongListFragment_;

/**
 * @author hoaht
 */
public class PagerAdapter extends FragmentStatePagerAdapter {
    private final String[] mTitles = {"Songs", "Albums", "Artist"};
    private final Fragment mSongListFragment = new SongListFragment_();
    private final Fragment mAlbumsFragment = new AlbumsFragment_();
    private final Fragment mArtistFragment = new ArtistFragment_();

    public PagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return mSongListFragment;
            case 1:
                return mAlbumsFragment;
            case 2:
                return mArtistFragment;
        }
        return null;
    }

    @Override
    public int getCount() {
        return mTitles.length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles[position];
    }
}

File: app/src/main/java/com/example/asiantech/playmusic/adapter/PagerAdapter.kt
package com.example.asiantech.playmusic.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import com.example.asiantech.playmusic.fragment.AlbumsFragment
import com.example.asiantech.playmusic.fragment.ArtistFragment
import com.example.asiantech.playmusic.fragment.SongListFragment

/**
 * PagerAdapter.
 *
 * @author HoaHT
 */
class PagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    private val titles = listOf("Songs", "Albums", "Artist")
    private val songListFragment = SongListFragment()
    private val albumsFragment = AlbumsFragment()
    private val artistFragment = ArtistFragment()

    override fun getItem(position: Int): Fragment? {
        when (position) {
            0 -> return songListFragment
            1 -> return albumsFragment
            2 -> return artistFragment
        }
        return null
    }

    override fun getCount(): Int {
        return titles.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }
}
